File: Laborator/Lab 3/JavaClientServer/Persistance/src/main/java/Repository/DBRepositoryEvent.java
package Repository;

import Domain.Event;
import Utils.DbUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.LinkedList;
import java.util.List;
import java.util.Properties;

public class DBRepositoryEvent implements IRepositoryEvent<Integer, Event> {
    private DbUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public DBRepositoryEvent(Properties props) {
        logger.info("Initializing repo props : {}",props);
        this.dbUtils = new DbUtils(props);
    }


    @Override
    public List<Event> findAll() {
        logger.traceEntry("finding all events");
        List<Event> l=new LinkedList<>();
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from Probes")){
            try(ResultSet result=preStmt.executeQuery()) {

                while (result.next()) {
                    int id = result.getInt("Id");
                    int min=result.getInt("AgeMin");
                    int max=result.getInt("AgeMax");
                    int distance=result.getInt("Distance");
                    Event e=new Event(id,min,max,distance);
                    l.add(e);
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        finally {

            try {
                con.close();
                logger.info("closed");
            } catch (SQLException ex) {
                logger.error(ex);
            }
        }
        if(l.isEmpty())
            logger.traceExit("No event");
        else
            logger.traceExit("Events are {}",l);
        return l;

    }

    @Override
    public Event find(Integer integer) {
        logger.traceEntry("finding event with id {} ",integer);
        Connection con=dbUtils.getConnection();
        Event e=null;
        try(PreparedStatement preStmt=con.prepareStatement("select * from Probes where Id=?")){
            preStmt.setInt(1,integer);
            try(ResultSet result=preStmt.executeQuery()) {

                if (result.next()) {
                    int id = result.getInt("Id");
                    int min=result.getInt("AgeMin");
                    int max=result.getInt("AgeMax");
                    int distance=result.getInt("Distance");
                    e=new Event(id,min,max,distance);
                    logger.traceExit("S-a gasit evntul {}",e);
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        finally {

            try {
                con.close();
                logger.info("closed");
            } catch (SQLException ex) {
                logger.error(ex);
            }
        }
        if(e.equals(null))
             logger.traceExit("No event found with id {}", integer);
        return e;
    }


}

File: Laborator/Lab 3/C# Client-server/Persistence/Repository/Database/DbRepositoryEvent.cs
using log4net;
using Model;
using Persistence.Utils;
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using MySql.Data.MySqlClient;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Persistence.Repository.Database
{
    public class DbRepositoryEvent : IRepositoryEvent<int, Event>
    {

        private DbUtils dbUtils;
        private static ILog logger = LogManager.GetLogger(typeof(DbRepositoryEvent));

        public DbRepositoryEvent(NameValueCollection properties)
        {
            dbUtils = new DbUtils(properties);
            logger.Info("Created new DbRepositoryEvent instance");
        }

        // TODO implement code
        public Event Find(int id)
        {
            logger.Debug("Finding event with id " + id + "...");

            MySqlConnection connection = (MySqlConnection)dbUtils.GetConnection();

            MySqlCommand selectCommand = new MySqlCommand("select * from Probes where Id=@idEv", connection);
            selectCommand.Parameters.Add("@idEv", MySqlDbType.Int32).Value = id;
            MySqlDataAdapter adapter = new MySqlDataAdapter(selectCommand);
            DataSet dataSet = new DataSet();

            adapter.Fill(dataSet);

            if (dataSet.Tables[0].Rows.Count == 0)
            {
                logger.Info("No event found with id " + id);
                return null;
            }

            logger.Debug("Event found with id " + id);
            return new Event()
            {
                ID = (int)dataSet.Tables[0].Rows[0]["Id"],
                Distance = (int)dataSet.Tables[0].Rows[0]["Distance"],
                AgeMin = (int)dataSet.Tables[0].Rows[0]["AgeMin"],
                AgeMax = (int)dataSet.Tables[0].Rows[0]["AgeMax"],
                No = 0
            };
        }

        public List<Event> FindAll()
        {
            logger.Debug("Finding all events...");

            MySqlConnection connection = (MySqlConnection)dbUtils.GetConnection();

            MySqlCommand selectCommand = new MySqlCommand("select * from Probes", connection);
            MySqlDataAdapter adapter = new MySqlDataAdapter(selectCommand);
            DataSet dataSet = new DataSet();

            adapter.Fill(dataSet);

            var events = from DataRow row in dataSet.Tables[0].Rows
                    select new Event()
                    {
                        ID = (int)row["Id"],
                        Distance = (int)row["Distance"],
                        AgeMin = (int)row["AgeMin"],
                        AgeMax = (int)row["AgeMax"]
                    };


            logger.Debug("Successfully found all events");

            return events.ToList();
        }
    }
}
